Fixes

heychirag opened the pull request:
1. Fixed loading of google maps api
2. The fix @Kolpa did in #66 isn't right I guess. I should have fixed this when I had changed the system of units earlier.

Ceilican commented on Source/core/CarbonFootprintCore.js:
```diff
@@ -15,14 +15,15 @@ CarbonFootprintCore.prototype.computeFootprint = function(distance) {
+    footprint = footprint * 16;
+    carbonUnit = 'oz';
+  }
+    footprint = Math.round(footprint*100)/100;
```
Why "100"? Why a precision of 2 decimal places?

heychirag replied:
if there are 2 routes, one with CO2 emission of 1.50 kg and other with 2.49 kg, both will show CO2 emission of 2 kg if not rounded off to decimals.

The precision can also be decreased to 1 decimal place if you like :)

Ceilican replied:
But if we have a route with CO2 emission of "300.21 kg", I would rather see "300 kg" instead of "300.21 kg"?

So, if you would like to show more decimal places, do it only for small values of carbon emission, when those decimal places are not negligible.

Ceilican commented:
> The fix @Kolpa did in #66 isn't right I guess. I should have fixed this when I had changed the system of units earlier.

I think you are both partially wrong. :-)

Before @heychirag's change, we were using "g" ("grams"). @heychirag changed it to "kg", but apparently he forgot to change the defaults. Look here, for instance:

https://github.com/heychirag/CarbonFootprintGoogleMaps/blob/a308a1f263e1708cbeb0fa203976cfa1b366ccd1/Source/core/settings/ChromeSettingsProvider.js#L42

The default CO2 emission rate should be 217 grams per km. But now, with @heychirag 's new units, I supect it might be 217 kilograms per meter! That's 10^6 higher than it should be!

I think it is these wrong default values that motivated @kolpa to change the unit back to "g" ("grams").

The standard unit in the metric unit system is "kg". So, it's comprehensible that @heychirag wants to use "kg" instead of "g". On the other hand, we should also be careful with the storage of values that have too many decimal places.

In the end, I don't care so much if we use "g" or "kg", but I do care that the defaults are consistent with the units we are using.

@heychirag , could you thoroughly check all default values and units used throughout the system and check that we are consistently using the same units everywhere? I will leave this PR open until you either  do all the changes that need to be done or tell me that you are sure that no more changes are needed.

heychirag commented:
I get your point. Actually it is 217 kg/km. Though meter is the SI unit, I have not taken it as the default unit since it is too small. A default value of 0.217 should fix this right?

Ceilican commented:
> I get your point. Actually it is 217 kg/km. Though meter is the SI unit, I have not taken it as the default unit since it is too small. A default value of 0.217 should fix this right?

Yes, a default of 0.217 would be right, if the unit is "kg/km".  (By the way, I found this arbitrary value of 217g/km in 2011. Nowadays the average emission rate for cars might be lower than this. If you find a reliable source with a different average emission rate, feel free to change our default value.)

> I get your point, floating points are not good for precision. I'll fix this right away.

Yes, although "0.217" is not so bad. Just keep this issue in mind, if you are going to calculate things with very small floating point numbers.

heychirag commented:
This branch is a lot of commits behind the master, therefore I am closing this PR. I'll add more commits on top of latest commits and raise a PR soon.

heychirag commented:
Hi @Ceilican,

>@heychirag , could you thoroughly check all default values and units used throughout the system and check that we are consistently using the same units everywhere?

I think I was finally able to fix all the issues. I have also fixed calculation of CO2 emission for different fuels that are measured by volume, mass and energy.

>Nowadays the average emission rate for cars might be lower than this. If you find a reliable source with a different average emission rate, feel free to change our default value.)

I might have found a reliable source but its interesting to note that emission rate is more than the current value. This might be because [this](https://www.epa.gov/sites/production/files/2016-02/documents/420f14040a.pdf) report covers only the vehicles inside the US only. For now, I have changed the default value to the one in this report. Let me know if we would like to keep this.

Ceilican commented:
Thanks!!

Ceilican commented on Source/core/settings/ChromeSettingsProvider.js:
```diff
@@ -69,7 +69,7 @@ ChromeSettingsProvider.prototype.get = function(key, def) {
 
 ChromeSettingsProvider.prototype.getCarbonEmission = function() {
-  return this.get('emissionRate', 217);
+  return this.get('emissionRate', 0.255384);
```
Could you add a comment with a link to the report?
